### DISCUSSION TITLE
Dependency Injection

### DISCUSSION THREAD

**Discussion by Wulfheart** [filamentphp/filament#5427] [Help] [answered]
Hello,

I would like to inject some things (mainly repositories and action classes to be triggered by filament actions). I didn't find anything in the docs. Can please give me some guidance please?

My current approach is using 
```php
/** @var BanUserFromGameAction $action */
$action = app(BanUserFromGameAction::class);
$action->execute($player);
```

But my approach feels tedious and unnecessarily bloated.

Thank you in advance.

**Comment by danharrin** [comment, marked as answer]
Every single closure in Filament supports DI

```php
function (BanUserFromGameAction $action, Player $record) {
    $action->execute($record);
}
```

**Reply by Wulfheart** [reply]
This is cool. It should definitely be part of the docs.
